### PR TITLE
Add a return value in ForeachDirectoryEntry 

### PR DESCRIPTION
**Pull request by LFsWang**
I find there is a mistake in #1256. Line of return was deleted and not rewrite it.

### CODE REVIEW

File: src/common/file_util.cpp [hunk 1]
```diff
@@ -427,6 +427,9 @@ bool ForeachDirectoryEntry(unsigned* num_entries_out, const std::string &directo
     // How many files + directories we found
     unsigned found_entries = 0;
 
+    // save the status of callback function
+    bool is_all_callback_successful = true;
```
Comment (archshift) [inline src/common/file_util.cpp]: Can you rename this to something more succinct, like `callback_error` or something, and adjust its usage as appropriate?

**Comment by archshift** [comment]
Nice catch!

Pretty much LGTM, although could you get rid of those two merge commits by rebasing please?

**Comment by LFsWang** [comment]
Ok. I renamed var and remove unnecessary merges

**Comment by LFsWang** [comment]
AppVeyor is still pending..

**Comment by kitling** [comment]
@LFsWang The AppVeyor build was canceled...? So it's never gonna complete. No idea why it was canceled.

Does anybody know of a way to get AppVeyor to re-check something?

**Comment by Subv** [comment]
Force-push to the branch

**Comment by LFsWang** [comment]
I pushed again. It works!

**Comment by bunnei** [comment]
LGTM

**Pull request merged**